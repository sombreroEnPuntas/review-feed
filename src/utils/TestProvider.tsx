import React from 'react'
import { Provider } from 'react-redux'
import { ThemeWrapper } from 'retro-ui'

import store from '../data'

const TestProvider = ({ children }) => (
  <ThemeWrapper>
    <Provider store={store}>{children}</Provider>
  </ThemeWrapper>
)

export default TestProvider
