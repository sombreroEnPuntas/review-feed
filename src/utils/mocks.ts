import {
  ModelJWTResponse,
  ModelAPIError,
  ModelTheme,
  ModelReview,
} from '../client/api'

export const internalError = new Error(`Wut?. SyntaxError or NetworkError`)

export const errorAPIResponse: ModelAPIError = {
  code: 418,
  message: `The resulting entity body MAY be short and stout.`,
}

export const unauthorizedAPIResponse: ModelAPIError = {
  code: 401,
  message: `You shall not pass!`,
}

export interface MockData {
  login?: ModelJWTResponse
  themes?: ModelTheme[]
  reviews?: ModelReview[]
}
export class MockService {
  data
  error

  constructor(data: MockData, error: 'internal' | 'API' | 'EXPIRED_TOKEN') {
    this.data = data
    this.error = error
  }

  private genericErrorResponse = () => {
    if (this.error === 'API') {
      throw { json: () => Promise.resolve(errorAPIResponse) }
    } else if (this.error === 'EXPIRED_TOKEN') {
      throw { json: () => Promise.resolve(unauthorizedAPIResponse) }
    } else if (this.error === 'internal') {
      throw internalError
    }
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  loginPost = async (_username: string, _password: string) => {
    this.genericErrorResponse()

    return this.data?.login
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  apiThemesGet = async (_limit: number, _offset: number) => {
    this.genericErrorResponse()

    return { data: this.data?.themes }
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  apiReviewsGet = async (_limit: number, _offset: number, _themeId: number) => {
    this.genericErrorResponse()

    return { data: this.data?.reviews }
  }
}

export const reviewMock = {
  id: 59435909,
  created_at: '2020-08-20T11:54:52Z',
  comment: 'Good, Bad, and Ugly.',
  themes: [
    { theme_id: 1000, sentiment: 1 },
    { theme_id: 1001, sentiment: 0 },
    { theme_id: 1002, sentiment: -1 },
  ],
}

export const themeListMock = [
  { id: 1000, name: 'General' },
  { id: 1001, name: 'Other' },
  { id: 1002, name: 'Unspecific' },
]
