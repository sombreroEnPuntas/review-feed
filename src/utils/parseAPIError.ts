import { ModelAPIError } from '../client/api'
import { clearToken } from '../data/login'

const UNAUTHORIZED_ERROR_CODE = 401

const parseAPIError = async (e: unknown, dispatch) => {
  let message: string

  // `instanceof Response` is not reliable when contexts are different :)
  // duct-taping is ok tho cuz we just need to unwrap a server response
  if ((e as Body).json) {
    const errorResponse: ModelAPIError = await (e as Body).json()
    if (errorResponse.code === UNAUTHORIZED_ERROR_CODE) {
      dispatch(clearToken())
    }
    message = `${errorResponse.code}: ${errorResponse.message}`
  } else {
    message = `${(e as Error).message}`
  }

  return message
}

export default parseAPIError
