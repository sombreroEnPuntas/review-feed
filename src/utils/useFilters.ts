import { useDispatch, useSelector } from 'react-redux'

import { ModelTheme } from '../client/api'
import { getThemeFilters, setThemeFilters } from '../data/filters'
import { getThemes } from '../data/themes'

const useFilters = () => {
  const themes: ModelTheme[] = useSelector(getThemes)
  const themeFilters: number[] = useSelector(getThemeFilters)

  const dispatch = useDispatch()

  const filter = (selection: number[]) => dispatch(setThemeFilters(selection))
  const reset = () => dispatch(setThemeFilters([]))

  return { filter, reset, themes, themeFilters }
}

export default useFilters
