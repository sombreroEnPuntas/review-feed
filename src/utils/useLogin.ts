import { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  clearToken,
  getError,
  getToken,
  initialToken,
  setToken,
} from '../data/login'

const useLogin = () => {
  const dispatch = useDispatch()

  const error: string = useSelector(getError)
  const token: string = useSelector(getToken)

  useEffect(() => {
    dispatch(initialToken())
  }, [dispatch])

  const login = (username: string, password: string) => {
    dispatch(setToken(username, password))
  }

  const logout = () => {
    dispatch(clearToken())
  }

  return { error, login, logout, token }
}

export default useLogin
