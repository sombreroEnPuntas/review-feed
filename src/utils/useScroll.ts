import { useState, useEffect } from 'react'

const useScroll = (callback) => {
  const [loading, setLoading] = useState(false)
  const [scroll, setScroll] = useState(0)

  function handleScroll() {
    if (
      window.innerHeight + document.documentElement.scrollTop !==
        document.documentElement.offsetHeight ||
      loading
    )
      return

    setScroll(document.documentElement.scrollTop)
    setLoading(true)
  }

  useEffect(() => {
    window.addEventListener('scroll', handleScroll)

    return function cleanUp() {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  useEffect(() => {
    if (!loading) return
    ;(async () => {
      await callback()
      setLoading(false)
      document.documentElement.scrollTop = scroll
    })()
  }, [loading])
}

export default useScroll
