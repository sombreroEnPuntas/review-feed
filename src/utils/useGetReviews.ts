import { useRef, useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { DefaultApi, ModelReview, ModelTheme } from '../client/api'
import { Configuration } from '../client/configuration'
import parseAPIError from './parseAPIError'
import { getToken } from '../data/login'

interface ModelThemeResponse {
  data: ModelTheme[]
}

interface ModelReviewResponse {
  data: ModelReview[]
}

const useGetReviews = () => {
  const service = useRef<DefaultApi>()

  const dispatch = useDispatch()
  const token: string = useSelector(getToken)

  const [error, setError] = useState<string>()
  const [loading, setLoading] = useState(false)
  const [reviews, setReviews] = useState<ModelReview[]>()
  const [themes, setThemes] = useState<ModelTheme[]>()

  const getThemes = async ({
    limit,
    offset,
  }: {
    limit?: number
    offset?: number
  }) => {
    setLoading(true)
    let message = null

    try {
      // wrong types on the model :(
      const response: unknown = await service.current.apiThemesGet(
        limit,
        offset
      )
      setThemes((response as ModelThemeResponse).data)
    } catch (e) {
      message = await parseAPIError(e, dispatch)
    }

    setError(message)
    setLoading(false)
  }

  const getReviews = async ({
    limit = 10,
    offset,
    themeId,
  }: {
    limit?: number
    offset?: number
    themeId?: number
  } = {}) => {
    setLoading(true)
    let message = null

    try {
      // wrong types on the model :(
      const response: unknown = await service.current.apiReviewsGet(
        limit,
        offset,
        themeId
      )
      setError(null)
      setReviews((prevData) => [
        ...(prevData || []),
        ...(response as ModelReviewResponse).data,
      ])
    } catch (e) {
      message = await parseAPIError(e, dispatch)
    }

    setError(message)
    setLoading(false)
  }

  // instantiate the service when new tokens come only
  useEffect(() => {
    ;(async () => {
      if (!token) return

      const config: Configuration = { apiKey: `Bearer ${token}` }
      service.current = new DefaultApi(config)

      if (!themes) {
        // initializes all, since it's just text
        await getThemes({ limit: 100 })
      }
      if (!reviews) {
        await getReviews()
      }
    })()

    return function cleanUp() {
      service.current = undefined
    }
  }, [token])

  return { error, loading, reviews, themeList: themes, getReviews }
}

export default useGetReviews
