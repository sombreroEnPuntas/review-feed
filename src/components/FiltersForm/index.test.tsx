import { render, fireEvent } from '@testing-library/react'
import React from 'react'

import TestedComponent from '.'

// Deps
import useFilters from '../../utils/useFilters'

// Utils
import { themeListMock } from '../../utils/mocks'
import { ThemeWrapper } from 'retro-ui'

// Mocks
jest.mock('../../utils/useFilters')

// Mock data
const setMock = ({ filter, reset, themes, themeFilters }) =>
  (useFilters as jest.Mock).mockImplementation(() => ({
    filter,
    reset,
    themes,
    themeFilters,
  }))

const stageTest = ({
  filter = jest.fn(),
  reset = jest.fn(),
  themes = themeListMock,
  themeFilters = [],
}) => {
  setMock({ filter, reset, themes, themeFilters })

  return render(
    <ThemeWrapper>
      <TestedComponent />
    </ThemeWrapper>
  )
}

describe('FiltersForm', () => {
  afterEach(() => {
    // much force-casting, such type-safe, so dark side
    ;(useFilters as jest.Mock).mockClear()
  })
  afterAll(() => {
    jest.resetAllMocks() // clean .mock
  })

  it(`renders main CTAs when not opened`, async () => {
    const testData = {}
    const { container, getByText } = stageTest(testData)

    const filterButton = getByText(/Filter/)
    const resetButton = getByText(/Reset/)

    expect(filterButton).toBeInTheDocument()
    expect(resetButton).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it(`renders a list of filters when opened`, async () => {
    const testData = {}
    const { container, getByText } = stageTest(testData)

    const filterButton = getByText(/Filter/)
    fireEvent.click(filterButton)

    const applyButton = getByText(/Apply/)
    const theme1 = getByText(/General/)
    const theme2 = getByText(/Other/)
    const theme3 = getByText(/Unspecific/)

    expect(applyButton).toBeInTheDocument()
    expect(theme1).toBeInTheDocument()
    expect(theme2).toBeInTheDocument()
    expect(theme3).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it(`reacts to query input`, async () => {
    const filter = jest.fn()
    const testData = { filter }
    const { container, getByText, getByLabelText } = stageTest(testData)

    const filterButton = getByText(/Filter/)
    fireEvent.click(filterButton)

    const queryInput = getByLabelText(/query/)
    fireEvent.change(queryInput, { target: { value: 'gene' } })

    const theme1 = getByText(/General/)

    expect(theme1).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it(`reacts to apply CTA`, async () => {
    const filter = jest.fn()
    const testData = { filter }
    const { container, getByText } = stageTest(testData)

    const filterButton = getByText(/Filter/)
    fireEvent.click(filterButton)

    const theme1 = getByText(/General/)
    fireEvent.click(theme1)

    const theme2 = getByText(/Other/)
    fireEvent.click(theme2)
    fireEvent.click(theme2) //unselect!

    const applyButton = getByText(/Apply/)
    fireEvent.click(applyButton)

    expect(filter).toHaveBeenCalledWith([1000])
    expect(container).toMatchSnapshot()
  })

  it(`reacts to reset CTA`, async () => {
    const reset = jest.fn()
    const testData = { reset }
    const { container, getByText, getAllByText } = stageTest(testData)

    const filterButton = getByText(/Filter/)
    fireEvent.click(filterButton)

    const theme1 = getByText(/General/)
    fireEvent.click(theme1)

    const resetButton = getAllByText(/Reset/)
    fireEvent.click(resetButton[0])
    fireEvent.click(resetButton[1])

    expect(reset).toHaveBeenCalledTimes(2)
    expect(container).toMatchSnapshot()
  })
})
