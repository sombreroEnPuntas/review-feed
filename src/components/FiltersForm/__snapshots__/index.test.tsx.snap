// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersForm reacts to apply CTA 1`] = `
.c0 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
  background-color: #ADACB1;
  box-shadow: none;
}

<div>
  <button
    class="c0"
  >
    Filter
  </button>
  <button
    class="c0"
  >
    Reset
  </button>
</div>
`;

exports[`FiltersForm reacts to query input 1`] = `
.c2 {
  background-color: white;
  color: #212529;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  overflow-x: auto;
  padding: 0.5rem 1rem;
  white-space: pre-wrap;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c9 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c7 {
  border: none;
  font-family: 'Press Start 2P',monospace;
  line-height: 1.5em;
  outline-color: #212529;
  padding: 0.5rem 1rem;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4d525666;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

.c6 {
  margin: 1em 0 0 0;
}

.c0 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
  background-color: #ADACB1;
  box-shadow: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 20em;
  overflow: auto;
}

.c5 {
  background-color: white;
  cursor: pointer;
  padding: 0 1px;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

<div>
  <button
    class="c0"
  >
    Filter
  </button>
  <button
    class="c0"
  >
    Reset
  </button>
  <div
    class="c1"
  >
    <div
      class="c2"
      type="black"
    >
      <form
        class="c3"
      >
        <div
          class="c2"
          type="black"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              General
            </div>
          </div>
        </div>
        <div
          class="c6"
        />
        <input
          aria-label="query"
          class="c7"
          name="query"
          placeholder="find theme..."
          type="text"
          value="gene"
        />
        <div
          class="c6"
        />
        <div
          class="c8"
        >
          <button
            class="c9"
            type="submit"
          >
            Apply
          </button>
          <button
            class="c9"
            type="button"
          >
            Reset
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`FiltersForm reacts to reset CTA 1`] = `
.c2 {
  background-color: white;
  color: #212529;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  overflow-x: auto;
  padding: 0.5rem 1rem;
  white-space: pre-wrap;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c9 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c7 {
  border: none;
  font-family: 'Press Start 2P',monospace;
  line-height: 1.5em;
  outline-color: #212529;
  padding: 0.5rem 1rem;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4d525666;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

.c6 {
  margin: 1em 0 0 0;
}

.c0 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
  background-color: #ADACB1;
  box-shadow: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 20em;
  overflow: auto;
}

.c5 {
  background-color: white;
  cursor: pointer;
  padding: 0 1px;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

<div>
  <button
    class="c0"
  >
    Filter
  </button>
  <button
    class="c0"
  >
    Reset
  </button>
  <div
    class="c1"
  >
    <div
      class="c2"
      type="black"
    >
      <form
        class="c3"
      >
        <div
          class="c2"
          type="black"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              General
            </div>
            <div
              class="c5"
            >
              Other
            </div>
            <div
              class="c5"
            >
              Unspecific
            </div>
          </div>
        </div>
        <div
          class="c6"
        />
        <input
          aria-label="query"
          class="c7"
          name="query"
          placeholder="find theme..."
          type="text"
          value=""
        />
        <div
          class="c6"
        />
        <div
          class="c8"
        >
          <button
            class="c9"
            type="submit"
          >
            Apply
          </button>
          <button
            class="c9"
            type="button"
          >
            Reset
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`FiltersForm renders a list of filters when opened 1`] = `
.c2 {
  background-color: white;
  color: #212529;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  overflow-x: auto;
  padding: 0.5rem 1rem;
  white-space: pre-wrap;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c9 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c7 {
  border: none;
  font-family: 'Press Start 2P',monospace;
  line-height: 1.5em;
  outline-color: #212529;
  padding: 0.5rem 1rem;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4d525666;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

.c6 {
  margin: 1em 0 0 0;
}

.c0 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
  background-color: #ADACB1;
  box-shadow: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 20em;
  overflow: auto;
}

.c5 {
  background-color: white;
  cursor: pointer;
  padding: 0 1px;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

<div>
  <button
    class="c0"
  >
    Filter
  </button>
  <button
    class="c0"
  >
    Reset
  </button>
  <div
    class="c1"
  >
    <div
      class="c2"
      type="black"
    >
      <form
        class="c3"
      >
        <div
          class="c2"
          type="black"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              General
            </div>
            <div
              class="c5"
            >
              Other
            </div>
            <div
              class="c5"
            >
              Unspecific
            </div>
          </div>
        </div>
        <div
          class="c6"
        />
        <input
          aria-label="query"
          class="c7"
          name="query"
          placeholder="find theme..."
          type="text"
          value=""
        />
        <div
          class="c6"
        />
        <div
          class="c8"
        >
          <button
            class="c9"
            type="submit"
          >
            Apply
          </button>
          <button
            class="c9"
            type="button"
          >
            Reset
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`FiltersForm renders main CTAs when not opened 1`] = `
.c0 {
  background-color: white;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  padding: 0.5em;
  box-shadow: 0 4px #4d5256,0 -4px #4d5256,4px 0 #4d5256,-4px 0 #4d5256;
  margin: 4px;
  background-color: #ADACB1;
  box-shadow: none;
}

<div>
  <button
    class="sc-AxirZ c0"
  >
    Filter
  </button>
  <button
    class="sc-AxirZ c0"
  >
    Reset
  </button>
</div>
`;
